feat: expose power consumption for heating, dhw and total

tlinnet opened the pull request:
https://github.com/openviess/PyViCare/issues/664

Commit:
Adding sensors for heating.power.consumption.dhw, heating.power.consumption.heating and heating.power.consumption.total

Commit:
Adding test response file for Vitocal252A

Commit:
Sorted Vitocal252A.json with jq, according to README

Commit:
Implementing initial tests for Vitocal252A. Copied from Vitocal250A and adjusted

Commit:
Implemented test for Power Consumption Heating

Commit:
Implemented tests for Total power consumption

Commit:
Adding test for Power consumption for Domestic Hot Water

Commit:
Merge branch 'master' into core_issue_155695

Commit:
Align naming on tests

Commit:
Update of test for Vitocal252A, after ViCare API had Maintenance window 2025-11-18. That seems to have fixed some values

Commit:
Adding missing Properties for tests. Marked under '# heating - not yet used'

CFenner commented on tests/test_Vitocal252A.py:
Adding a test for the newly added data point would also have been enough to 😉

tlinnet replied:
Yeah... :)
But I had forgot to include the GazBoiler. And then in one of the responses, the Unit was hidden differently.
So, I went an extra mile to see if I found any surprises.
If it's too much, let me know, and I will carve it out again.

CFenner commented on PyViCare/PyViCareHeatPump.py:
```diff
@@ -51,6 +44,23 @@ def getBufferTopTemperature(self):
+        return self.getProperty("heating.power.consumption.heating")["properties"]["month"]["value"][0]
+
+    @handleNotSupported
+    def getPowerConsumptionHeatingYear(self):
```
```suggestion
    def getPowerConsumptionHeatingThisYear(self):
```

CFenner commented on tests/test_Vitocal252A.py:
```diff
@@ -0,0 +1,253 @@
+
+    def test_getPowerConsumptionHeatingYear(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionHeatingYear(), 263.2)
```
```suggestion
    def test_getPowerConsumptionHeatingThisYear(self):
        self.assertEqual(
            self.device.getPowerConsumptionHeatingThisYear(), 263.2)
```

CFenner replied:
As, this is a total mess, there are already method that are names this way, but also the other way 🤯

Commit:
Adding sensors to GazBoiler

Commit:
Adding another test for GazBoiler

Commit:
Fix for some GazBoilers have unit in another dict

Commit:
Added additional test for FuelCell

Commit:
Adding sensors to fuelcell

Commit:
adding extra test for Gaz boiler

Commit:
Adding additional tests

tlinnet committed:
Update PyViCare/PyViCareHeatPump.py

def getPowerConsumptionHeatingYear(self):
    def getPowerConsumptionHeatingThisYear(self):

Co-authored-by: Christopher Fenner <9592452+CFenner@users.noreply.github.com>

Commit:
Renaming getPowerConsumptionHeatingYear to getPowerConsumptionHeatingThisYear

Commit:
Fix for missing return, after implementation of fetching Unit of kilowattHour, that can be stored in either properties.day.unit or properties.unit.value

Commit:
Renamed getPowerConsumptionDomesticHotWaterYear to getPowerConsumptionDomesticHotWaterThisYear

tlinnet commented:
There was one thing I was unsure abouat

`grep "    def " PyViCareHeatPump.py | cut -d" " -f6 | cut -d"(" -f1 | sort | uniq -d` shows that some function definitions are made multiple times

```
getLiquidTemperature
getLiquidTemperatureUnit
getPowerConsumptionCoolingThisYear
getPowerConsumptionHeatingThisYear
```

Not sure if this could be an issue.
Test cases are running as expected

tlinnet committed:
Merge branch 'master' into core_issue_155695

tlinnet commented:
Merge from upstream

CFenner commented on PyViCare/PyViCareGazBoiler.py:
```diff
@@ -204,6 +209,26 @@ def getGasSummaryConsumptionDomesticHotWaterLastYear(self):
 
     # Power consumption for Heating:
     @handleNotSupported
+    def getPowerConsumptionHeatingUnit(self):
```
Is the fallback really necessary? Maybe it's just that the test data is outdated.

tlinnet replied:
The fallback was necessary, to make the test-suite complete.
If there is outdated test data, it's hard to fix it for me in this PR.

Commit:
pylint Fix C0304: Final newline missing (missing-final-newline) in test_Vitodens200W, test_Vitodens200W_B2HF, test_Vitodens222W, test_Vitodens300W

Commit:
pylint Fix C0303: Trailing whitespace (trailing-whitespace) in test_VitovalorPT2, line 110, column 20

tlinnet commented:
Hi @CFenner . 
What would be needed to move this forward?

tlinnet commented:
Hi @CFenner 

I tested the PR, by setting up `vicare` in `homeassistant/custom_components`, and adding the new sensors.
This seems now to work as expected

<img width="1032" height="1102" alt="image" src="https://github.com/user-attachments/assets/08d6f879-02fc-4bdc-a15a-b7218857a676" />

tlinnet commented:
Lets go :)
It's working as intended

tlinnet commented:
Anyone here?

lackas commented:
Hi @tlinnet, just a heads up that there are related power consumption efforts happening:

- **PR #667** (yours): `heating.power.consumption` dhw+heating+total methods
- **PR #688**: Vitocal 300-G support (COP, compressor power, refrigerant sensors, cooling circuits)
- **PR #689**: `heating.heatingRod.power.consumption.summary` methods (addresses issue #677)

All add power consumption and sensor features for heat pumps - complementary work with no conflicts. Mentioning in case you or the maintainers want to coordinate the reviews.

tlinnet commented:
Hi @CFenner 
Can this be merged ?

tlinnet commented:
Hi @lackas and @CFenner 
Cab we complete this PR before it turns into dust?

CFenner reviewed (approved):
👍
